File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash_map = {}

        for i, num in enumerate(nums):
            if num in hash_map:
                return [i, hash_map[num]]
            hash_map[target - num] = i
        

File: 0001-two-sum/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> numIndex = new Dictionary<int, int>();
        
        for (int i = 0; i < nums.Length; i++){
            if (numIndex.ContainsKey(target - nums[i])){
                return [i, numIndex[target - nums[i]]];
            }
            if (!numIndex.ContainsKey(nums[i])){
                numIndex.Add(nums[i], i);
            }
        }
        return [-1, -1];
    }
}